packages/node_modules/@ciscospark/media-engine-webrtc/test/unit/lib/offer-answer.js

```diff
@@ -1,107 +1,124 @@
 import {assert} from '@ciscospark/test-helper-chai';
-import {find} from 'lodash';
+import {find, filter} from 'lodash';
 import {parse} from 'sdp-transform';
+import {Promise} from 'es6-promise';
 
-let pc;
-export function mockAnswer(offer) {
-  const parsed = parse(offer);
+let pc, stream;
 
-  console.log(`mockAnswering an sdp with ${parsed.media.length} media sections`);
-  if (pc && pc.signalingState !== 'closed') {
-    pc.getSenders().forEach((s) => s.track && s.track.stop());
-    pc.close();
-  }
-  pc = new RTCPeerConnection();
-  return navigator.mediaDevices.getUserMedia({
-    audio: true,
-    video: true,
-    fake: true
-  })
-    .then((stream) => {
-      pc.addStream(stream);
-    })
-    .then(() => pc.setRemoteDescription({type: 'offer', sdp: offer}))
-    .then(() => pc.createAnswer())
-    .then((answer) => pc.setLocalDescription(answer)
-      .then(() => answer.sdp))
-    .then((answer) => {
-      const slides = find(parsed.media, {
-        content: 'slides',
-        type: 'video'
-      });
-      if (slides) {
-        const [a, b] = answer.split(`a=mid:${slides.mid}`);
-        answer = [a, `a=mid:${slides.mid}`, '\r\na=content:slides', b].join('');
-      }
-      console.log(`mockAnswering with ${parsed.media.length} media sections`);
-      return answer;
-    });
+export function setupRemotePeerConnection() {
+  console.log('resetting remote PC and stream');
+  pc = new RTCPeerConnection({
+    iceServers: [],
+    bundlePolicy: 'max-compat'
+  });
+
+  stream = null;
 }
+/**
+ * Simulates a server's sdp response
+ * @param {string} offerSdp created from engine.createOffer
+ * @returns {object} answer
+ */
+export function simulateAnswer(offerSdp) {
+  const sdp = parse(offerSdp);
+
+  const audioMedia = filter(sdp.media, {type: 'audio'});
+  const videoMedia = filter(sdp.media, {type: 'video'});
 
-export function mockRenegotiate(offer) {
-  const sdp = parse(offer);
-  console.log(`mockAnswering an sdp with ${sdp.media.length} media sections`);
+  const wantsAudio = audioMedia && !!find(audioMedia, (m) => m.direction.includes('recv'));
+  const wantsVideo = videoMedia && !!find(videoMedia, (m) => m.direction.includes('recv'));
 
-  const wantsAudio = find(sdp.media, {type: 'audio'}) && find(sdp.media, {type: 'audio'}).direction.includes('recv');
-  const wantsVideo = find(sdp.media, {type: 'video'}) && find(sdp.media, {type: 'video'}).direction.includes('recv');
+  const hasAudio = !!pc.getSenders().find((s) => s.track && s.track.kind === 'audio');
+  const hasVideo = !!pc.getSenders().find((s) => s.track && s.track.kind === 'video');
 
-  const hasAudio = !!pc.getLocalStreams()[0].getAudioTracks().length;
-  const hasVideo = !!pc.getLocalStreams()[0].getVideoTracks().length;
+  console.log(`wantsAudio: ${wantsAudio}`);
```

Comment:
We've got a mix of `.log` and `.info` should we standardize on one?

Reply:
Good call. I'm gonna update it and use `info` and then `warn` if it's something bad.